diff --git a/Makefile b/Makefile
index 138ace1..c296331 100644
--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@ DIRECTORIES = \
   $(TOOLKIT)/Rewind
 
 ifeq ($(OS), Linux)
-  FLAGS += -rdynamic
+  FLAGS += -static
   KIND := $(shell grep -E "^6.0" /etc/debian_version > /dev/null ; echo $?)
 ifneq ($(KIND), 0)
   LIBRARIES += rt
@@ -26,7 +26,7 @@ OBJECTS = \
 FLAGS += -g -fno-omit-frame-pointer -O3 -MMD $(foreach directory, $(DIRECTORIES), -I$(directory)) -DBUILD=\"$(BUILD)\"
 LIBS += $(foreach library, $(LIBRARIES), -l$(library))
 
-CC = gcc
+CC ?= gcc
 CFLAGS += $(FLAGS) -std=gnu99
 
 ifneq ($(strip $(DEPENDENCIES)),)
