From b3250320e516bc66a63ceb2890a381631205fc6b Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Thu, 20 Jun 2024 20:21:21 +0000
Subject: [PATCH] add CMakeLists.txt

---
 CMakeLists.txt | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..1ed7fa0
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,32 @@
+cmake_minimum_required(VERSION 3.13)
+
+set(APP_NAME "DAPNETGateway")
+project(${APP_NAME} CXX)
+
+if(GIT_VERSION)
+  set(GIT_HASH ${GIT_VERSION})
+else()
+  execute_process(
+    COMMAND
+      git log -1 --pretty=format:%h --abbrev=10 HEAD ${CMAKE_SOURCE_DIR}
+    OUTPUT_VARIABLE GIT_HASH
+  )
+  if (NOT GIT_HASH)
+    set(GIT_HASH "0000000000")
+  endif()
+endif()
+message(STATUS "Git revision: " ${GIT_HASH})
+
+file(WRITE ${CMAKE_SOURCE_DIR}/GitVersion.h "const char *gitversion = \"${GIT_HASH}\";")
+
+file(GLOB SOURCES "*.cpp")
+file(GLOB HEADERS "*.h")
+
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Os -Wall -std=c++0x -DHAVE_LOG_H $ENV{CXXFLAGS}")
+set(DEPLIBS "")
+
+find_package(Threads REQUIRED QUIET)
+list(APPEND DEPLIBS ${CMAKE_THREAD_LIBS_INIT})
+
+add_executable(${APP_NAME} ${SOURCES} ${HEADERS})
+target_link_libraries(${APP_NAME} ${DEPLIBS})
-- 
2.39.2

